### ISSUE TITLE
Add methods to clear connection pools

### ISSUE THREAD

**Issue by bgrainger** [bgrainger/MySqlConnector#46]
MySql.Data provides the following APIs on `MySqlConnection`:

```
public static void ClearPool(MySqlConnection connection)
public static void ClearAllPools()
public Task ClearPoolAsync(MySqlConnection connection)
public Task ClearPoolAsync(MySqlConnection connection, CancellationToken cancellationToken)
public Task ClearAllPoolsAsync()
public Task ClearAllPoolsAsync(CancellationToken cancellationToken)
```

It's not clear to me why the sync methods are static method, while the async methods are instance methods (that sometimes take a `MySqlConnection` object as a parameter).

Nevertheless, for compatibility we should probably clone this API as-is (and deprecate `MySqlHelper` in this library).

API documentation:

* [ClearPool](https://dev.mysql.com/doc/dev/connector-net/html/M_MySql_Data_MySqlClient_MySqlConnection_ClearPool.htm)
* [ClearAllPools](https://dev.mysql.com/doc/dev/connector-net/html/M_MySql_Data_MySqlClient_MySqlConnection_ClearAllPools.htm)
* [ClearPoolAsync](https://dev.mysql.com/doc/dev/connector-net/html/Overload_MySql_Data_MySqlClient_MySqlConnection_ClearPoolAsync.htm)
* [ClearAllPoolsAsync](https://dev.mysql.com/doc/dev/connector-net/html/Overload_MySql_Data_MySqlClient_MySqlConnection_ClearAllPoolsAsync.htm)

**Comment by bgrainger** [comment]
It could be that omitting `static` was an oversight in MySql.Data. Certainly it would be more useful for `MySqlConnection.ClearAllPoolsAsync` to be `static`, and porting existing code would not be difficult if the API were deliberately different in this library. (We could even add an `[Obsolete]` instance method that forwarded calls to the static method, for binary compatibility.)

**Comment by bgrainger** [comment]
Shipped in [0.1.0-alpha17](https://www.nuget.org/packages/MySqlConnector/0.1.0-alpha17).

**Comment by bgrainger** [comment]
Added `static` implementations; consumers coming from MySql.Data (and using the async methods) will need to update their code to call the static methods instead of instance ones.
